app/coffee/Router.coffee

```diff
@@ -79,14 +79,25 @@ module.exports = Router =
 					if err?
 						Router._handleError null, err, client, "leaveProject"
 
-
-			client.on "joinDoc", (doc_id, fromVersion, callback) ->
-				# fromVersion is optional
+			# Variadic. The possible arguments:
+			# doc_id, callback
+			# doc_id, fromVersion, callback
+			# doc_id, options, callback
+			# doc_id, fromVersion, options, callback
+			client.on "joinDoc", (doc_id, fromVersion, options, callback) ->
 				if typeof fromVersion == "function"
```

Comment:
Sorry to be nitpicky here... I'm finding the if statements below hard to reason about. Could we be really explicit here?

```
if typeof fromVersion == 'function' and !options?

else if typeof fromVersion == 'object' and typeof object == 'function'

else if typeof fromVersion == 'number' and typeof object == 'function'

else if typeof fromVersion == 'number' and typeof options == 'object'
   # pass, things are as expected
else
   logger.error "unexpect arguments"
   callback ( new Error(...) )
```

Reply:
No that's a good point. I had to think to reason it out